### DISCUSSION TITLE
Best way to add mailto address or URL to author field in an html doc

### DISCUSSION THREAD

**Discussion by AlexMcClung** [quarto-dev/quarto-cli#540] [Q&A] [answered]
Hello,

What is the best way to add a clickable URL or mailto link in the author field in YAML in an html doc? Probably kludgey, but with Rmd I used to add an email as a footnote with: 

---
author: 
  - My Office Name^[email@email.edu] 
---

With quarto html rendered via knitr I get the footnote okay but the author text has the caret and email in brackets.

Thanks in advance for your help!

And loving Quarto, thanks for all of your work on this!!

**Comment by cderv** [comment, marked as answer]
Hi, 

With this 
````markdown
---
title: test
author: My Office Name^[email@email.edu]
format: html
---

## Content

```{r}
1 + 1 
```
````

I don't see what you are talking about with last dev version, but that is not better
![image](https://user-images.githubusercontent.com/6791940/161932262-1636d2e0-2fcd-4c52-bc50-39ce8b9accae.png)

We have a double footnote and the hover 
![image](https://user-images.githubusercontent.com/6791940/161932443-d2cbd9e3-af9a-4633-990a-3caec77f5795.png)

Do you have an example to run with what you described ? 

Also I am using last dev version of Quarto 

> What is the best way to add a clickable URL or mailto link in the author field in YAML

You need to add `<url>` for this which is the syntax for url
````
author: My Office Name^[<email@email.edu>]
````

This is the expected syntax for links (https://pandoc.org/MANUAL.html#automatic-links)


Otherwise you need to explicitly activate the extensions [`autolink_bare_uri`](https://pandoc.org/MANUAL.html#extension-autolink_bare_uris) if you want any URL to become a link
````yaml
title: test
author: My Office Name^[email@email.edu]
format: 
  html:
    from: markdown+autolink_bare_uris
````

This is not a default in Quarto (it is in R Markdown, hence the different behavior)

**Reply by AlexMcClung** [reply]
Thanks Christophe! My mistake, I was running quarto version 0.9.83 on windows. Once I updated it to version 0.9.188 I was able to get the double footnote and the text for author properly resolves to "My Office Name" with the superscript rather than containing the caret, brackets, etc. as it did before. 

Also, I did not need to add the extension or use pointy brackets. It worked fine for me with just the caret and the bare url in square brackets.

**Reply by cderv** [reply]
Glad it works. 
@dragonstyle the double footnotes does not seem right in the screenshot above. Do you agree ? Maybe you have some pointers ?

**Reply by dragonstyle** [reply]
Yes, I agree! I think the issue is caused by the fact that we sometimes inject front matter elements into the markdown to be rendered (this is necessary for markdown content that pandoc might not see and so would otherwise be underendered). In this case, I think we are likely doing this, but unnecessarily, and pandoc is also rendering the author (and so there are two footnotes in the markdown that is rendered, the author as it normally would be, and the front matter payload that we have incorrectly injected for rendering).

This is a bug in Quarto that I should be able to quickly fix!

**Reply by dragonstyle** [reply]
One other note- you can currently just write:

```
author:
  - name: My Name
  - url: mailto:email@email.edu
```

And this will correctly be rendered as a link For HTML output (though it will not use a footnote). I took a look at improving our rendering to avoid the double footnote, and two things jumped out at me:

1) It is actually a little more involved than I initially though since you can also specify affiliations via the author yaml, and we need a we need a way to keep all this data together ourselves for rendering as a unit.

2) In the long run, you'll be better served by keeping the structure of the data separated from the presentation. I'm currently working on improving our templating, author, and title block support and I think expressing structured author data will be very helpful as it will be utilized by a variety of different templates and rendering (e.g. everyone can have their own way of rendering the author's email)

I'm going to keep pushing forward on the improved templating and table fixing this for now (it will be fixed when the new templating work lands soon). But if you can live with the way we render the email address for now, I think we'll have better support for what you're looking for soon...

(edit: corrected syntax above)
